From efe8636c8345bac78791e2b5f57f5a630578046e Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Wed, 18 Nov 2015 19:37:42 -0800
Subject: [PATCH] try <asm/sgidefs.h> as well as <sigmips.h>

Hack for MIPS builds with musl libc, where sgidefs.h isn't found in
the expected location?

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 configure.ac | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index e73958c..4d8692f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -158,7 +158,19 @@ if test "$arch" = mips; then
 			[AC_LANG_PROGRAM([[#include <sgidefs.h>]],
 					 [[int i[_MIPS_SIM == _MIPS_SIM_ABI64 ? 1 : - 1];]])],
 			[ac_cv_mips_abi=n64],
-			[ac_cv_mips_abi=unknown])])])])
+		[AC_COMPILE_IFELSE(
+			[AC_LANG_PROGRAM([[#include <asm/sgidefs.h>]],
+					 [[int i[_MIPS_SIM == _MIPS_SIM_ABI32 ? 1 : - 1];]])],
+			[ac_cv_mips_abi=o32],
+		[AC_COMPILE_IFELSE(
+			[AC_LANG_PROGRAM([[#include <asm/sgidefs.h>]],
+					 [[int i[_MIPS_SIM == _MIPS_SIM_NABI32 ? 1 : - 1];]])],
+			[ac_cv_mips_abi=n32],
+		[AC_COMPILE_IFELSE(
+			[AC_LANG_PROGRAM([[#include <asm/sgidefs.h>]],
+					 [[int i[_MIPS_SIM == _MIPS_SIM_ABI64 ? 1 : - 1];]])],
+			[ac_cv_mips_abi=n64],
+			[ac_cv_mips_abi=unknown])])])])])])])
 	case "$ac_cv_mips_abi" in
 		o32) AC_DEFINE([LINUX_MIPSO32], [1], [Define for _MIPS_SIM_ABI32.]);;
 		n32) AC_DEFINE([LINUX_MIPSN32], [1], [Define for _MIPS_SIM_NABI32.]);;
-- 
1.9.1

